state_transfer/transfer/rsync/client.go

```diff
@@ -106,3 +136,30 @@ func createRsyncClient(c client.Client, r *RsyncTransfer, ns string) error {
 
 	return errorsutil.NewAggregate(errs)
 }
+
+// customizeTransportClientContainers customizes transport's client containers for specific rsync communication
+func customizeTransportClientContainers(t transport.Transport) {
+	switch t.Type() {
+	case stunnel.TransportTypeStunnel:
+		stunnelContainer := &t.ClientContainers()[0]
```

Comment:
instead of keying on t.ClientContainers()[0], can we key on t.ClientContainers()[*].Name, this will avoid bugs when refactoring later on